rest_framework/views.py

```diff
@@ -414,6 +415,11 @@ def finalize_response(self, request, response, *args, **kwargs):
             response.accepted_media_type = request.accepted_media_type
             response.renderer_context = self.get_renderer_context()
 
+        if 'Vary' in self.headers:
+            # Add new vary headers to existing ones instead of overwriting.
+            new_vary_headers = cc_delim_re.split(self.headers.pop('Vary'))
+            patch_vary_headers(response, new_vary_headers)
+
```

Comment:
Okay, I'm convinced.

Let's tweak the code block slightly tho. The `pop` is hidden away inside the rest of the code so it's not totally clear at first why it doesn't jsut later get overridden.

How about...

    vary = self.headers.pop('Vary', None)
    if vary is not None:
        ...